tests/scripts/local_monitoring_tests/rest_apis/test_create_experiment.py

```diff
@@ -0,0 +1,294 @@
+from helpers.kruize import *
+from helpers.utils import *
+from jinja2 import Environment, FileSystemLoader
+
+mandatory_fields = [
+    ("version", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("cluster_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("experiment_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("mode", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("target_cluster", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("kubernetes_objects", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("type", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("kubernetes_objects_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("namespace", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("containers", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("container_image_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("container_name", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("selector", SUCCESS_STATUS_CODE, SUCCESS_STATUS),
+    ("namespace", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("performance_profile", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("slo", SUCCESS_STATUS_CODE, SUCCESS_STATUS),
+    ("recommendation_settings", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("trial_settings", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("kubernetes_objects_name_selector", ERROR_STATUS_CODE, ERROR_STATUS),
+    ("performance_profile_slo", ERROR_STATUS_CODE, ERROR_STATUS)
+]
+
+csvfile = "/tmp/create_exp_test_data.csv"
+
+@pytest.mark.sanity
+def test_create_namespace_exp_with_namespace_type(cluster_type):
+    """
+    Test Description: This test validates the response status code of createExperiment API
+    for namespace experiment by passing a valid input for the json
+    """
+    input_json_file = "../json_files/create_namespace_exp.json"
+    form_kruize_url(cluster_type)
+
+    response = delete_experiment(input_json_file)
+    print("delete exp = ", response.status_code)
+
+    # Create experiment using the specified json
+    response = create_experiment(input_json_file)
+
+    data = response.json()
+    print(data['message'])
+
+    assert response.status_code == SUCCESS_STATUS_CODE
+    assert data['status'] == SUCCESS_STATUS
+    assert data['message'] == CREATE_EXP_SUCCESS_MSG
+
+    response = delete_experiment(input_json_file)
+    print("delete exp = ", response.status_code)
+
+@pytest.mark.negative
+def test_create_namespace_exp_without_type(cluster_type):
+    """
+    Test Description: This test validates the response status code of createExperiment API
+    for namespace experiment by passing a invalid input for the json without specifying the experiment type
+    """
+    input_json_file = "../json_files/create_namespace_exp_without_type.json"
+    form_kruize_url(cluster_type)
+
+    response = delete_experiment(input_json_file)
+    print("delete exp = ", response.status_code)
+
+    # Create experiment using the specified json
+    response = create_experiment(input_json_file)
+
+    data = response.json()
+    print(data['message'])
+
+    assert response.status_code == ERROR_500_STATUS_CODE
```

Comment:
Can you include validation for the error message too in all the invalid tests.  And should the error code be 500 or 400?

Reply:
It's 500 for now. I have noted it, we can include validation for the proper error message.